app/upgrades_test.go

```diff
@@ -647,3 +652,103 @@ func (s *UpgradeTestSuite) TestRemoveInactiveValidatorDelegations() {
 		s.Assert().Len(validators, 3, "GetAllValidators after %s", runnerName)
 	})
 }
+
+func (s *UpgradeTestSuite) TestSetNewGovParamsTestnet() {
+	var runErr error
+	runner := func() {
+		runErr = setNewGovParamsTestnet(s.ctx, s.app)
+	}
+	runnerName := "setNewGovParamsTestnet"
+
+	threeDays := time.Hour * 24 * 3
+	fiveMinutes := time.Minute * 5
+
+	iniParams := govv1.DefaultParams()
+	iniParams.MinInitialDepositRatio = sdkmath.LegacyMustNewDecFromStr("0.25").String()
+	iniParams.MinDepositRatio = sdkmath.LegacyMustNewDecFromStr("0.10").String()
+	iniParams.ProposalCancelRatio = sdkmath.LegacyMustNewDecFromStr("0.83").String()
+	iniParams.ProposalCancelDest = sdk.AccAddress("addr________________").String()
+	iniParams.ExpeditedVotingPeriod = &threeDays
+	iniParams.ExpeditedThreshold = sdkmath.LegacyMustNewDecFromStr("0.406").String()
+	iniParams.ExpeditedMinDeposit = []sdk.Coin{sdk.NewInt64Coin("banana", 1000)}
+	iniParams.BurnVoteQuorum = true
+	iniParams.BurnProposalDepositPrevote = true
+	iniParams.BurnVoteVeto = false
+
+	err := s.app.GovKeeper.Params.Set(s.ctx, iniParams)
+	s.Require().NoError(err, "Setting initial gov params")
+
+	expParams := govv1.DefaultParams()
+	expParams.MinInitialDepositRatio = sdkmath.LegacyMustNewDecFromStr("0.00002").String()
+	expParams.MinDepositRatio = sdkmath.LegacyZeroDec().String()
+	expParams.ProposalCancelRatio = sdkmath.LegacyZeroDec().String()
+	expParams.ProposalCancelDest = ""
+	expParams.ExpeditedVotingPeriod = &fiveMinutes
+	expParams.ExpeditedThreshold = sdkmath.LegacyMustNewDecFromStr("0.667").String()
+	expParams.ExpeditedMinDeposit = iniParams.MinDeposit
+	expParams.BurnVoteQuorum = false
+	expParams.BurnProposalDepositPrevote = false
+	expParams.BurnVoteVeto = true
+
+	expLogLines := []string{
+		"INF Setting new gov params for testnet.",
+		"INF Done setting new gov params for testnet.",
+	}
+
+	s.ExecuteAndAssertLogs(runner, expLogLines, nil, true, runnerName)
+	s.Require().NoError(runErr, runnerName)
+
+	actParams, err := s.app.GovKeeper.Params.Get(s.ctx)
+	s.Require().NoError(err, "getting gov params after %s", runnerName)
+	s.Assert().Equal(expParams, actParams, "resulting gov params")
+}
```

Comment:
Unfortunately, because of decisions that the SDK has made, there's no way to mock what's needed in order to trigger the negative test cases. This is the best we can do right now.